crates/uv/tests/pip_compile.rs

```diff
@@ -4225,3 +4225,77 @@ fn override_with_incompatible_constraint() -> Result<()> {
 
     Ok(())
 }
+
+/// Resolve a package with a strict upper bound, allowing pre-releases. Per PEP 440, pre-releases
+/// that match the bound (e.g., `2.0.0rc1`) should be _not_ allowed.
+#[test]
+fn pre_release_upper_bound_exclude() -> Result<()> {
+    let context = TestContext::new("3.12");
+    let requirements_in = context.temp_dir.child("requirements.in");
+    requirements_in.write_str("flask<2.0.0")?;
+
+    uv_snapshot!(context.compile()
+            .arg("requirements.in")
+            .arg("--prerelease=allow"), @r###"
+    success: true
+    exit_code: 0
+    ----- stdout -----
+    # This file was autogenerated by uv via the following command:
+    #    uv pip compile --cache-dir [CACHE_DIR] --exclude-newer 2023-11-18T12:00:00Z requirements.in --prerelease=allow
+    click==7.1.2
+        # via flask
+    flask==1.1.4
+    itsdangerous==1.1.0
+        # via flask
+    jinja2==2.11.3
+        # via flask
+    markupsafe==2.1.3
+        # via jinja2
+    werkzeug==1.0.1
+        # via flask
+
+    ----- stderr -----
+    Resolved 6 packages in [TIME]
+    "###
+    );
+
+    Ok(())
+}
+
+/// Resolve a package with a strict upper bound that includes a pre-release. Per PEP 440,
+/// pre-releases that match the bound (e.g., `2.0.0rc1`) should be _not_ allowed.
+#[test]
+fn pre_release_upper_bound_include() -> Result<()> {
+    let context = TestContext::new("3.12");
+    let requirements_in = context.temp_dir.child("requirements.in");
+    requirements_in.write_str("flask<2.0.0rc4")?;
+
+    uv_snapshot!(context.compile()
+            .arg("requirements.in")
+            .arg("--prerelease=allow"), @r###"
+    success: true
+    exit_code: 0
+    ----- stdout -----
+    # This file was autogenerated by uv via the following command:
+    #    uv pip compile --cache-dir [CACHE_DIR] --exclude-newer 2023-11-18T12:00:00Z requirements.in --prerelease=allow
+    click==8.1.7
+        # via flask
+    flask==2.0.0rc2
+    itsdangerous==2.1.2
+        # via flask
+    jinja2==3.1.2
+        # via flask
+    markupsafe==2.1.3
+        # via
+        #   jinja2
+        #   werkzeug
+    werkzeug==3.0.1
+        # via flask
+
+    ----- stderr -----
+    Resolved 6 packages in [TIME]
+    "###
+    );
```

Comment:
Is this test case correct? The comment seems to say something different.

Reply:
Sorry, the comment is wrong.

Reply:
(`flask<2.0.0rc4` _should_ allow `2.0.0rc2`.)